sunpy/coordinates/tests/test_metaframes.py

```diff
@@ -100,7 +100,7 @@ def test_as_base(rot_hgs):
     # Check the as_base() method
     a = rot_hgs[1].as_base()
 
-    assert type(a) == type(rot_hgs[1].base)
+    assert isinstance(a, rot_hgs[1].base)
```

Comment:
```suggestion
    assert type(a) == type(rot_hgs[1].base)  # NOQA
```
This intentionally checks that the type is strictly the same.

Reply:
I often do `typea is typeb` rather than `typea == typeb` I wonder if there is a good reason to do one over the other :thinking:

Reply:
For me, I'd use `is` if the right operand is a "literal" (e.g., `type(a) is astropy.coordinates.ICRS`).  Here, with both operands being expressions, `==` seems more appropriate.  Maybe I'm just weird.

Reply:
I would rather say that using `is` or `==` is a matter of testing [whether you have the same object](https://docs.python.org/3/reference/expressions.html#is) or whether you have the same value of the object. In this case, `type()` applied to different objects of the same type always returns the same `type` object (not just different objects with the same value), so I would stick to `is`, but I don't have a really good argument for not using `==` (which also works).